src/Microsoft.Sbom.Api/Config/Validators/ManifestInfoValidator.cs

```diff
@@ -17,17 +18,38 @@ namespace Microsoft.Sbom.Api.Config.Validators;
 /// </summary>
 public class ManifestInfoValidator : ConfigValidator
 {
-    public ManifestInfoValidator(IAssemblyConfig assemblyConfig)
-        : base(typeof(ValidManifestInfoAttribute), assemblyConfig)
+    private readonly HashSet<ManifestInfo> supportedManifestInfos;
+
+    /// <summary>
+    /// Production constructor (satisfied by dependency injection)
+    /// </summary>
+    public ManifestInfoValidator(IAssemblyConfig assemblyConfig, ManifestGeneratorProvider manifestGeneratorProvider)
+        : this(assemblyConfig, GetAvailableManifestInfos(manifestGeneratorProvider))
+    {
+    }
+
+    /// <summary>
+    /// Test constructor
+    /// </summary>
+    public ManifestInfoValidator(IAssemblyConfig assemblyConfig, HashSet<ManifestInfo> supportedManifestInfos)
+    : base(typeof(ValidManifestInfoAttribute), assemblyConfig)
     {
+        this.supportedManifestInfos = supportedManifestInfos ?? throw new ArgumentNullException(nameof(supportedManifestInfos));
     }
 
     public override void ValidateInternal(string paramName, object paramValue, Attribute attribute)
     {
-        if (paramValue is not null && paramValue is List<ManifestInfo> listOfManifestInfos && !Constants.SupportedSpdxManifests.Any(listOfManifestInfos.Contains))
+        if (paramValue is not null && paramValue is List<ManifestInfo> listOfManifestInfos && !supportedManifestInfos.Any(listOfManifestInfos.Contains))
         {
-            var supportedManifests = string.Join(", ", Constants.SupportedSpdxManifests.Select(m => m.ToString()));
-            throw new ValidationArgException($"Please provide a valid value for the ManifestInfo (-mi) parameter. Supported values include: {supportedManifests}. The values are case-insensitive.");
+            var providedValues = string.Join(", ", listOfManifestInfos);
+            var validManifestInfoa = string.Join(", ", supportedManifestInfos.Select(m => m.ToString()));
+            throw new ValidationArgException($"The value '{providedValues}' contains no values supported by the ManifestInfo (-mi) parameter. Please provide supported values. Supported values include: {validManifestInfoa}. The values are case-insensitive.");
         }
     }
+
+    private static HashSet<ManifestInfo> GetAvailableManifestInfos(ManifestGeneratorProvider manifestGeneratorProvider)
+    {
+        ArgumentNullException.ThrowIfNull(manifestGeneratorProvider, nameof(manifestGeneratorProvider));
+        return [.. manifestGeneratorProvider.GetSupportedManifestInfos()];
```

Comment:
Would GetSupportedManifestInfos() also get supported manifest info values through dependency injection? For example, would this call return WdacCatalogManifestInfo that's injected in drop validator?